openlibrary/plugins/upstream/addbook.py

```diff
@@ -68,6 +69,18 @@ def is_plugin_enabled(name):
     else:
         return None
 
+def force_invalidate_cache(docs):
+    """
+    Force invalidate documents from the cache
+    Temporary fix to https://github.com/internetarchive/openlibrary/issues/2040
+    :param list of (web.storage or dict) docs: documents to remove from the cache
+    :return: None
+    """
+    docs = [dict(doc) for doc in docs if doc]
+    keys = [doc['key'] for doc in docs if doc.has_key('key')]
+    keys += ["d" + k for k in keys]
```

Comment:
I don't fully understand the purpose of these 'd' prefixed keys, but they seem to include all the revisions, not just the current revision, and have been, as far as I have seen, been correctly updated.

Reply:
Yeah, that's it exactly. They store "metadata" about a document (like its history). I'll add a comment.

Reply:
```suggestion
    keys += ["d" + k for k in keys]  # also clear doc 'metadata'
```